web/crux/src/app/deploy/deploy.mapper.ts

```diff
@@ -292,7 +292,8 @@ export default class DeployMapper {
       initContainers: this.mapInitContainerToAgent(config.initContainers),
       portRanges: config.portRanges,
       ports: config.ports,
-      user: config.user,
+      // Set user to the given value, if not null or use 0 if specifically 0, otherwise set to default: -1
+      user: config.user || config.user === 0 ? (config.user === 0 ? 0 : config.user) : -1,
```

Comment:
This should be like: if the `config.user` is -1 then it's null otherwise `config.user`, but just to be sure check the agent's grpc mapper.

Reply:
Yea, this is the reason why the tests are failing.